### PR TITLE
Closes #55 Percentiles for age were being miscalculated

### LINKED ISSUE #55: Bug: PCTL isn't calculated properly for age-charts if BMI_correction is set to FALSE
### What happened?

I missed a spot in fixing the functions last week

### Session Information

_No response_

### Reproducible Example

_No response_

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralpeds (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralpeds site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_params_growth_age.R [hunk 1]
```diff
@@ -283,6 +283,10 @@ derive_params_growth_age <- function(dataset,
         ) %>%
         select(-c(P95, Sigma))
     }
+    else {
+      add_pctl <- add_pctl %>%
+        mutate(AVAL = pnorm(AVAL)*100)
```
Comment (zdz2101) [inline R/derive_params_growth_age.R:288]: @rossfarrugia I originally had the code set up was 

1. calculate z-score
2. use pnorm to calculated percentile
3. if bmi correction, use AVAL to recalculated appropriate percentile (this is incorrect cuz aval is currently already in percentile form 

so I modified it to 
1. get z-score
2. calculate bmi correction for percentile
3. !!! forgot to still calculate the percentile (This is what the code is now doing)

Comment (zdz2101) [reply R/derive_params_growth_age.R:288]: that said, the numbers for percentile are finally looking correct again, I was so focused on checking that the bmi correction was doing done properly I forgot to check that it was running normally again

**Comment by zdz2101** [comment]
image

image

@rossfarrugia Just to point it out, the patient with abnormally low head circumference even a sanity check with the WHO excel makes sense too, data prior to their "428 days old" ~14 month old visit, you see their HC from the 359-407 visits is well below 40, as we can see on the WHO's HC for age chart, even the P01 percentile indicates for that age ~12-13 months old, the P01 percentile is 40.7-41

**Comment by rossfarrugia** [comment]
@Fanny-Gautier @Lina2689 would you have chance to sanity check this update and test run it with your template to see all looks good?
FYI @Minlei0201 too as your issue is the unit tests for this one.

**Comment by Fanny-Gautier** [comment]
> @Fanny-Gautier @Lina2689 would you have chance to sanity check this update and test run it with your template to see all looks good?

I'll try to find some times tomorrow. thanks

**Comment by rossfarrugia** [comment]
@zdz2101 does this PR resolve the following issue @Minlei0201 sent by email?

During the testing, I noticed the `set_values_to_pctl` parameter derivation in the `derive_params_growth_age` function is incorrect - it still derives SDS instead of the percentile.

**Comment by Fanny-Gautier** [comment]
> @zdz2101 does this PR resolve the following issue @Minlei0201 sent by email?
> 
> During the testing, I noticed the `set_values_to_pctl` parameter derivation in the `derive_params_growth_age` function is incorrect - it still derives SDS instead of the percentile.

I think so, I can see the following difference:
- Before : 
image
- After:
image

**Comment by zdz2101** [comment]
> > @zdz2101 does this PR resolve the following issue @Minlei0201 sent by email?
> 
> > 
> 
> > During the testing, I noticed the `set_values_to_pctl` parameter derivation in the `derive_params_growth_age` function is incorrect - it still derives SDS instead of the percentile.
> 
> 
> 
> I think so, I can see the following difference:
> 
> - Before : 
> 
> image
> 
> - After:
> 
> image
> 
> 

Yes the pctl's should all be correctly calculated here now

**Review by Fanny-Gautier — COMMENTED** [review]
double checked and the PCTLs properly implemented now
